plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchange.java

```diff
@@ -78,14 +78,14 @@
     private final CompletableFuture<List<ExchangeSourceHandle>> exchangeSourceHandlesFuture = new CompletableFuture<>();
 
     public FileSystemExchange(
-            URI baseDirectory,
+            List<URI> baseDirectories,
             FileSystemExchangeStorage exchangeStorage,
             ExchangeContext exchangeContext,
             int outputPartitionCount,
             Optional<SecretKey> secretKey,
             ExecutorService executor)
     {
-        this.baseDirectory = requireNonNull(baseDirectory, "baseDirectory is null");
+        this.baseDirectories = requireNonNull(ImmutableList.copyOf(baseDirectories), "baseDirectories is null");
```

Comment:
`ImmutableList.copyOf(requireNonNull(baseDirectories, "baseDirectories is null"))`